web/src/components/zonemap.jsx

```diff
@@ -70,15 +71,22 @@ const ZoneMap = ({
   };
 
   // Generate two sources (clickable and non-clickable zones), based on the zones data.
+  // The `sources` object will trigger a whole re-rendering of the map, and will
+  // thus re-render all zones.
+  // This is a slower process than `setFeatureState`.
   const sources = useMemo(() => {
-    const features = Object.entries(zones).map(([zoneId, zone]) => {
+    // We here iterate over the zones list (instead of dict) to keep the iteration
+    // order stable
+    const features = zoneValues.map((zone, i) => {
       const length = (coordinate) => (coordinate ? coordinate.length : 0);
+      const zoneId = zone.config.countryCode;
       return {
         type: 'Feature',
         geometry: {
           ...zone.geography.geometry,
           coordinates: zone.geography.geometry.coordinates.filter(length), // Remove empty geometries
         },
+        id: i, // assign an integer id so the feature can be updated later on
```

Comment:
by setting an id (which needs to be an integer) we can later have a lightweight operation that updates the features by querying by id.